File: Baekjun/8.3/1149. RGB거리/solution.swift
let N = Int(readLine()!)!
var dp:[[Int]] = Array(repeating: Array(repeating: 0, count: 3), count: N)
let abc = readLine()!.split(separator: " ").compactMap {Int($0)}

dp[0][0] = abc[0]
dp[0][1] = abc[1]
dp[0][2] = abc[2]


for i in 1...N-1 {
    let abc = readLine()!.split(separator: " ").compactMap {Int($0)}

    dp[i][0] = min(dp[i-1][1], dp[i-1][2]) + abc[0]
    dp[i][1] = min(dp[i-1][0], dp[i-1][2]) + abc[1]
    dp[i][2] = min(dp[i-1][1], dp[i-1][0]) + abc[2]
}

print(dp.last!.min()!)

File: Baekjun/8.3/1149. RGB거리/solution.py
N = int(input())
dp = [[0] * 3 for _ in range(N)]

a, b, c = map(int, input().split())

dp[0][0] = a
dp[0][1] = b
dp[0][2] = c

for i in range(1, N):
    a, b, c = map(int, input().split())
    dp[i][0] = min(dp[i - 1][1], dp[i - 1][2]) + a
    dp[i][1] = min(dp[i - 1][0], dp[i - 1][2]) + b
    dp[i][2] = min(dp[i - 1][0], dp[i - 1][1]) + c

print(min(dp[-1]))
